html/mycakeapp/src/Controller/MypageController.php

```diff
@@ -45,6 +53,124 @@ public function delete()
         $this->Auth->logout();
         return $this->redirect(['controller' => 'members', 'action' => 'deleted']);
     }
+
+    public function reserved()
+    {
+        $this->viewBuilder()->setLayout('frame-title');
+        $memberId = $this->Auth->user('id');
+        $week = ['日', '月', '火', '水', '木', '金', '土'];
+        $reserved = null;
+        //現在以降にログインユーザーが予約している情報を抽出
+        $reservedLists = $this->Payments->find('ReservedLists', ['memberId' => $memberId]);
+        //ログインユーザーが予約している情報をスケジュールごとに分け、連想配列を作成
+        $i = '0';
+        $j = 0;
+        foreach ($reservedLists as $reservedList) {
+            $detailContents =  array(
+                'movieTitle' => $reservedList->schedule->movie->name,
+                'moviePictureName' => $reservedList->schedule->movie->picture_name,
+                'date' => $reservedList->schedule->start_date->format('n月j日'),
+                'week' => $week[$reservedList->schedule->start_date->format('w')],
+                'startTime' => $reservedList->schedule->start_date->format('G:i'),
+                'finishTime' => date('G:i', strtotime('+' . $reservedList->schedule->movie->screening_time . 'minute', strtotime($reservedList->schedule->start_date))),
+                'column_number' => $reservedList->column_number,
+                'record_number' => $reservedList->record_number,
+                'payment' => $reservedList->purchase_price,
+                'scheduleId' => $reservedList->schedule_id
+            );
+            if ($reservedList->schedule_id === $i) {
+                $detail = $detailContents;
+                if ($reservedList->reservation_detail->discount_id !== 0) {
+                    $discount = $this->Discounts->get($reservedList->reservation_detail->discount_id);
+                    $detail['discountName'] = $discount->name;
+                }
+                $reserved[$j][] = $detail;
+                unset($detail);
+            } elseif ($reservedList->schedule_id !== $i) {
+                $detail = $detailContents;
+                if ($reservedList->reservation_detail->discount_id !== 0) {
+                    $discount = $this->Discounts->get($reservedList->reservation_detail->discount_id);
+                    $detail['discountName'] = $discount->name;
+                }
```

Comment:
滅茶苦茶スッキリしましたね！！
if文の中で共通に出ている箇所を抜き出すとより良いと思います！！
```suggestion
            $detail = $detailContents;
            if ($reservedList->reservation_detail->discount_id !== 0) {
                $discount = $this->Discounts->get($reservedList->reservation_detail->discount_id);
                $detail['discountName'] = $discount->name;
            }
            if ($reservedList->schedule_id === $i) {
                $reserved[$j][] = $detail;
                unset($detail);
            } elseif ($reservedList->schedule_id !== $i) {
```

Reply:
超スッキリしました！！
なるほど！確かにまとめられる！
修正します！